src/version/get-versions.ts

```diff
@@ -79,25 +79,20 @@ export function getOldestVersions(
   packageName: string,
   token: string
 ): Observable<VersionInfo[]> {
-  return queryForNewestVersions(
-    owner,
-    repo,
-    packageName,
-    token
-  ).pipe(
+  return queryForNewestVersions(owner, repo, packageName, token).pipe(
     map(result => {
       if (result.repository.packages.edges.length < 1) {
         throwError(
           `package: ${packageName} not found for owner: ${owner} in repo: ${repo}`
-        )
+        );
       }
 
       const versions = result.repository.packages.edges[0].node.versions.edges;
 
       return versions
-        .filter(value => value.node.version.indexOf(branchName) >= 0)
-        .map(value => ({id: value.node.id, version: value.node.version}))
-        .reverse()
+        .filter(value => value.node.version.includes(branchName))
```

Comment:
This line is what I added to filter the returned versions from the API to match branchName that we get from the input.